x/deployment/keeper/keeper.go

```diff
@@ -159,16 +176,17 @@ func (k Keeper) WithDeployments(ctx sdk.Context, fn func(types.Deployment) bool)
 	}
 }
 
-// WithDeploymentsActive filters to only those with State: Active
-func (k Keeper) WithDeploymentsActive(ctx sdk.Context, fn func(types.Deployment) bool) {
+// WithDeploymentState filters to only those with parameterized State
+func (k Keeper) WithDeploymentState(ctx sdk.Context, d types.DeploymentState, fn func(types.Deployment) bool) {
+	kvPrefix, err := deploymentStateKey(d)
+	if err != nil {
+		panic(err)
```

Comment:
`error` returned, however in the freak chance that an error was returned from WithDeploymentsInState(), in the `OnEndBlock()`; then what? Should that panic because the block closing failed?

Reply:
yeah, let's limit panics to `OnEndBlock`.